packages/scheduler/npm/index.js

```diff
@@ -1,6 +1,8 @@
 'use strict';
 
-if (process.env.NODE_ENV === 'production') {
+if (typeof window === 'undefined' || typeof MessageChannel !== 'function') {
+  module.exports = require('./unstable_no_dom');
```

Comment:
I'm not sure about this pattern because it's the first time we start doing this kind of switching in the npm wrappers and it won't port nicely to ESM.

We should move this branching into the implementation which forces us to treat this nicely in an ESM world.